src/django_idom/components.py

```diff
@@ -83,62 +67,121 @@ async def async_renderer():
+                view_html = view_or_template_view
 
         # Render Check 4: Sync class view
         elif getattr(view, "as_view", None):
             async_cbv = database_sync_to_async(view.as_view())
             view_or_template_view = await async_cbv(request_obj, *args, **kwargs)
             if getattr(view_or_template_view, "render", None):  # TemplateView
-                render = await database_sync_to_async(view_or_template_view.render)()
+                view_html = await database_sync_to_async(view_or_template_view.render)()
             else:  # View
-                render = view_or_template_view
+                view_html = view_or_template_view
 
         # Render Check 5: Sync function view
         else:
-            render = await database_sync_to_async(view)(request_obj, *args, **kwargs)
+            view_html = await database_sync_to_async(view)(request_obj, *args, **kwargs)
 
         # Signal that the view has been rendered
-        set_rendered_view(
+        set_converted_view(
             utils.html_to_vdom(
-                render.content.decode("utf-8").strip(),
+                view_html.content.decode("utf-8").strip(),
                 *transforms,
                 strict=strict_parsing,
             )
         )
 
-    # Return the view if it's been rendered via the `async_renderer` hook
-    return rendered_view
+    # Return the view if it's been rendered via the `async_render` hook
+    return converted_view
+
+
+# TODO: Might want to intercept href clicks and form submit events.
+# Form events will probably be accomplished through the upcoming DjangoForm.
+def view_to_component(
+    view: Callable | View = None,  # type: ignore[assignment]
+    compatibility: bool = False,
+    transforms: Sequence[Callable[[VdomDict], Any]] = (),
+    strict_parsing: bool = True,
+) -> Callable[[HttpRequest | None, Sequence | None, Dict | None], Component]:
+    """Converts a Django view to an IDOM component.
+
+    Keyword Args:
+        view: The view function or class to convert.
+        compatibility: If True, the component will be rendered in an iframe.
+            When using compatibility mode `tranforms`, `strict_parsing`, `request`,
+            `args, and `kwargs` arguments will be ignored.
+        transforms: A list of functions that transforms the newly generated VDOM.
+            The functions will be called on each VDOM node.
+        strict_parsing: If True, an exception will be generated if the HTML does not
+            perfectly adhere to HTML5.
+
+    Returns:
+        Callable: A function that takes a request, args, and kwargs and returns an IDOM
+            component.
+    """
+
+    def decorator(view: Callable | View):
+        if not view:
+            raise ValueError("A view must be provided to `view_to_component`")
+
+        def wrapper(
+            request: HttpRequest | None = None,
+            args: Sequence | None = None,
+            kwargs: Dict | None = None,
```

Comment:
Why not make this `*args, **kwargs`? Seems like a better user experience than passing a tuple and dict directly.

Reply:
No way to type hint `request, *args, **kwargs` properly.

Reply:
It's valid to annotate this as `request: HttpRequest | None = None, *args: Any, **kwargs: Any`

Reply:
Yes but what about [line 113](https://github.com/Archmonger/django-idom/blob/ba93fcd5fbd8d2bd6c7022fb4d5c712165d165ea/src/django_idom/components.py#L113)?

Reply:
It's ugly, but you'd annotate that using a protocol:

```python
class _ViewComponentConstructor(Protocol):
    def __call__(self, request: HttpRequest, *args: Any, **kwargs: Any) -> ComponentType:
        ...

def view_to_component(f: Callable, ...) -> _ViewComponentConstructor: ...
```